sql/sql_table.cc

```diff
@@ -9611,17 +9612,25 @@ enum fk_column_change_type
                                  change in foreign key column.
   @retval FK_COLUMN_RENAMED      Foreign key column is renamed.
   @retval FK_COLUMN_DROPPED      Foreign key column is dropped.
+  @retval FK_COLUMN_NOT_NULL     Foreign key column cannot be null.
 */
 
 static enum fk_column_change_type
-fk_check_column_changes(THD *thd, Alter_info *alter_info,
-                        List<LEX_CSTRING> &fk_columns,
-                        const char **bad_column_name)
+fk_check_column_changes(THD *thd, const TABLE *table,
+                        Alter_info *alter_info,
+                        const FOREIGN_KEY_INFO *fk,
+                        const char **bad_column_name,
+                        bool referenced=false)
 {
+  List<LEX_CSTRING> fk_columns= referenced
+      ? fk->referenced_fields
+      : fk->foreign_fields;
```

Comment:
Could this be as follows:
```c++
  const List<LEX_CSTRING> &fk_columns= referenced
    ? fk->referenced_fields
    : fk->foreign_fields;
```
That is, do not copy the list, but have a `const` reference to it.